mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+
+    def __include_test_users(self):
+        """
+        Includes one test user per available role.
+        """
+        new_user_mock = {
+            "notes": "I am test user :)",
+            "seen_notifications": [],
+            "pwg": [],
+            "email": "example@example.com",
+            "history": [],
+        }
+        new_users: list[dict] = []
+        all_roles = [r.value for r in Roles]
+
+        for role in all_roles:
+            new_user = deepcopy(new_user_mock)
+            new_user["_id"] = role
+            new_user["username"] = role
+            new_user["role"] = role
+            new_user["fullname"] = f"Test user for role: {role}"
+            new_users += [new_user]
+
+        # Send the request.
+        self.couchdb_request(
+            endpoint="users/_bulk_docs", method="POST", data={"docs": new_users}
+        )
+
+    def mcm_client(self, role: Roles) -> McM:
+        """
+        Return an instance of the McM client to perform requests
+        to the web application.
+
+        Args:
+            role (Roles): Role to impersonate in the requests.
+        """
+        return McM(api=self, role=role)
+
+
+class McM:
+    """
+    A simple version for McM REST client adapted to
+    be used for building the test cases.
+
+    Arguments:
+        api (APIRequest): Client to perform HTTP requests.
+        role (Roles): User role to impersonate when performing the
+            requests.
+    """
+
+    def __init__(self, api: APIRequest, role: Roles):
+        self.api: APIRequest = api
+        self.role: Roles = role
+        self.logger = self.__logger()
+
+    def __logger(self) -> logging.Logger:
+        """
+        Returns a logger for the class
+        """
+        logger = logging.getLogger("mcm-rest-client")
+        formatter = logging.Formatter("[%(asctime)s][%(levelname)s] %(message)s")
+        handler = logging.StreamHandler(sys.stdout)
+
+        handler.setFormatter(formatter)
+        handler.setLevel(logging.INFO)
+        logger.addHandler(handler)
+        return logger
+
+    def _get(self, url) -> tuple[dict, Response]:
+        response = self.api.mcm_request(endpoint=url, method="GET", as_role=self.role)
+        return response.json(), response
+
+    def _put(self, url, data) -> tuple[dict, Response]:
+        response = self.api.mcm_request(
+            endpoint=url, method="PUT", data=data, as_role=self.role
+        )
+        return response.json(), response
+
+    def _delete(self, url) -> tuple[dict, Response]:
+        response = self.api.mcm_request(
+            endpoint=url, method="DELETE", as_role=self.role
+        )
+        return response.json(), response
+
+    # McM methods
+    def get(
+        self, object_type, object_id=None, query="", method="get", page=-1
+    ) -> dict | None:
+        """
+        Get data from McM
+        object_type - [chained_campaigns, chained_requests, campaigns, requests, flows, etc.]
```

Comment:
Create an enumeration?

Reply:
Agree. I will check in the web application REST API what are all the possible options and pack them in an enum